test/Test/Pos/Lrc/WorkerSpec.hs

```diff
@@ -92,8 +94,12 @@ genTestParams = do
             (genAddressesAndDistrs totalStakeGroup (toList invSecretsMap))
             (enumerate allRichmenComponents)
     let _tpStakeDistributions = snd <$> addressesAndDistrs
-    let utxo = genesisUtxo Nothing addressesAndDistrs
-    let _tpGenTxpContext = mkGenesisTxpContext utxo
+    let genStakeholders =
+            GenesisWStakeholders $
+            HM.fromList $ map ((,1) . addressHash . toPublic) $
+            take 4 $ secretKeys
```

Comment:
What is it? How does it even work? I will check, I don't understand how tests can pass.

In any case, you should use `genWStakeholders` stakeholders here. Also I would wrap this combination of `genesisUtxo` and `genWStakeholders` into a single function because it seems quite useful.